src/llm/llm_manager.py

```diff
@@ -33,20 +32,40 @@ def invoke(self, prompt: str) -> str:
 class OpenAIModel(AIModel):
     def __init__(self, api_key: str, llm_model: str):
         from langchain_openai import ChatOpenAI
-        self.model = ChatOpenAI(model_name=llm_model, openai_api_key=api_key,
-                                temperature=0.4)
+
+        self.model = ChatOpenAI(
+            model_name=llm_model, openai_api_key=api_key, temperature=0.4
+        )
 
     def invoke(self, prompt: str) -> BaseMessage:
         logger.debug("Invoking OpenAI API")
         response = self.model.invoke(prompt)
         return response
 
 
+class AIMLModel(AIModel):
+    def __init__(self, api_key: str, llm_model: str):
+        from langchain_openai import ChatOpenAI
+
+        self.base_url = "https://api.aimlapi.com/v2"
```

Comment:
this endpoint

Reply:
Yes, the URL is correct, and I tested the basic functionality of the app—it was working.